BringYourOwnBoard.md

```diff
@@ -0,0 +1,157 @@
+
+### What do I need to think about? 
+
+For a customer looking at porting this section over there's not a lot you need to change. You need to edit the file such that it depends on your own custom image's bitbake file, edit the sw-description file to reflect what you would like the system to look like after an update, and provide tooling for accessing a private signing key. 
+
+## recipes-graphics
+
+### What does it do? 
+This just sets up graphics for the RPi. 
+### What do I need to think about? 
+There's nothing that affects customers in this section. You should just be able to entirely omit it from your port. 
+
+## recipes-support
+### What does it do? 
+
+In our project the [recipes-support](./recipes-support/) directory appends some of the extra tooling around swupdate. We use the [adu-swupdate-hw-compat.bb](./recipes-support/adu-swupdate-hw-compat/adu-swupdate-hw-compat.bb) to install a hardware compatibility file onto the image which swupdate looks for when completing an update. The version held within this file MUST match the one inside of the `sw-description` file mentioned above. 
+
+We also append swupdate with our own configuration inside of the [defconfig file](./recipes-support/swupdate/swupdate/raspberrypi4-64/defconfig). 
+### What do I need to think about? 
+
+For your `hw-compat` file that SwUpdates uses you should have a method for using either a hash of the hardware compatibilities, use a model number for the device you're deploying updates to, or really anything other than just a number that does not allow for any sort of granularity. The version held within this file MUST mach the one inside of the `sw-description` file mentioned above. 
+
+
+Depending on your use case your `defconfig` may be the same or different. You can read through the options provided by swupdate [here](https://sbabic.github.io/swupdate/swupdate-best-practise.html?highlight=defconfig#swupdate-builtin-configuration). Our configurations reflect the settings for our proof-of-concept. You will likely have different ones. That's ok. The file structure remains the same. Just make sure you edit the `*.swu` bitbake file to provide the items required for your configuration. 
+
+
+# wic
+
+### What does it do? 
+The `wic` directory provides our custom [adu-raspberrypi.wks file](./wic/adu-raspberrypi.wks) which uses the [wic](https://docs.yoctoproject.org/dev-manual/wic.html?highlight=wic#creating-partitioned-images-using-wic) tool within Open Embedded to generate the partitions required for the updates to run. 
+
+### What do I need to think about? 
+
+Your BSP may or may not support `wic`. If they do not you will need to use whatever tool they provide to accomplish the setup of the partitions. The only important consideration for porting is to make sure that you have the four required partitions: bootloader, partition A, partition B, and the ADU partition. 
```

Comment:
A BSP is singular (I think), so it should be "If it does not, you will need to...". Or say "BSP provider"?